### PR TITLE
CloudWatch: Use context in aws ListSinks and ListAttachedLinks

### PR DESCRIPTION
**Pull request by shabeebk**
**What is this feature?**

In the current way, ListSinks and ListAttachedLinks are used which doesn't allow canceling the request if the context changes. Using ListSinksWithContext and ListAttachedLinksWithContext is the preferred way.

**Why do we need this feature?**

Using context in the go-based application is always the preferred way wherever it is possible to do so. This is also related to the issue #75498.

**Who is this feature for?**

Users

**Which issue(s) does this PR fix?**:

Relates to #75498

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/tsdb/cloudwatch/mocks/oam_client.go [hunk 1]
```diff
@@ -9,12 +11,12 @@ type FakeOAMClient struct {
 	mock.Mock
 }
 
-func (o *FakeOAMClient) ListSinks(input *oam.ListSinksInput) (*oam.ListSinksOutput, error) {
+func (o *FakeOAMClient) ListSinksWithContext(ctx aws.Context, input *oam.ListSinksInput, opts ...request.Option) (*oam.ListSinksOutput, error) {
```
Comment (fridgepoet) [inline pkg/tsdb/cloudwatch/mocks/oam_client.go:14]: Doesn't the ctx need to be of type context.Context?

Comment (shabeebk) [reply pkg/tsdb/cloudwatch/mocks/oam_client.go:14]: Yes, it should be. I guess it got replaced due to my vscode env. Will fix and push a commit.

**Review by fridgepoet — APPROVED** [review]
Thanks for your contribution!
We have a CI issue, but we'll keep an eye on getting this merged.

**Comment by fridgepoet** [comment]
Hi again @shabeebk almost done. 
Would you git merge grafana's `main` branch into your branch again? Thanks!

**Comment by shabeebk** [comment]
Thanks for the notification @fridgepoet. Done.

**Comment by fridgepoet** [comment]
Hi again @shabeebk, the CI tests are still failing on a check that should **not** apply to you. We've updated that, but it's still failing, so I'm thinking it's because maybe your branch is not up-to-date with the latest changes in that CI (drone) file. Would you mind merging in the latest changes from the repo's `main` again?

**Comment by shabeebk** [comment]
Done @fridgepoet :)

**Pull request merged**